Model/AdyenInitiateTerminalApi.php

```diff
@@ -106,47 +106,69 @@ public function initiate()
-                            "MessageCategory": "Payment",
-                            "SaleID": "Magento2Cloud",
-                            "POIID": "' . $poiId . '",
-                            "ProtocolVersion": "3.0",
-                            "ServiceID": "' . $serviceID . '"
-                        },
-                        "PaymentRequest": {
-                            "SaleData": {
-                                "SaleTransactionID": {
-                                    "TransactionID": "' . $reference . '",
-                                    "TimeStamp": "' . $timeStamper . '"
-                                },
-                                "TokenRequestedType": "Customer"
-                            },
-                            "PaymentTransaction": {
-                                "AmountsReq": {
-                                    "Currency": "' . $quote->getCurrency()->getQuoteCurrencyCode() . '",
-                                    "RequestedAmount": ' . $quote->getGrandTotal() . '
-                                }
-                            },
-                            "PaymentData": {
-                                "PaymentType": "' . $transactionType . '"
-                            }
-                        }
-                    }
-                }
-            ';
-
-        $params = json_decode($json, true); //Create associative array for passing along
+        $customerId = $quote->getCustomerId();
+
+        $request = [
+            'SaleToPOIRequest' =>
+                [
+                    'MessageHeader' =>
+                        [
+                            'MessageType' => 'Request',
+                            'MessageClass' => 'Service',
+                            'MessageCategory' => 'Payment',
+                            'SaleID' => 'Magento2Cloud',
+                            'POIID' => $poiId,
+                            'ProtocolVersion' => '3.0',
+                            'ServiceID' => $serviceID,
+                        ],
+                    'PaymentRequest' =>
+                        [
+                            'SaleData' =>
+                                [
+                                    'TokenRequestedType' => 'Customer',
+                                    'SaleTransactionID' =>
+                                        [
+                                            'TransactionID' => $reference,
+                                            'TimeStamp' => $timeStamper,
+                                        ],
+                                ],
+                            'PaymentTransaction' =>
+                                [
+                                    'AmountsReq' =>
+                                        [
+                                            'Currency' => $quote->getCurrency()->getQuoteCurrencyCode(),
+                                            'RequestedAmount' => $quote->getGrandTotal(),
```

Comment:
(double)$quote->getGrandTotal() makes sure it is not converted to string by the json_encode

Reply:
double does not seems to exists can you check if doubleval() works as well ?
http://php.net/manual/en/function.doubleval.php